core/src/main/scala/io/projectglow/sql/expressions/LogisticRegressionGwas.scala

```diff
@@ -173,48 +168,169 @@ class NewtonArguments(
 }
 
 case class NewtonResult(
-    args: NewtonArguments,
+    args: NewtonIterationsState,
     logLkhd: Double,
     nIter: Int,
     converged: Boolean,
     exploded: Boolean)
 
 trait LogitTest extends Serializable {
+  type FitState
   def resultSchema: StructType
+  def canReuseNullFit: Boolean
+  def fitNullModel(phenotypes: Array[Double], covariates: SparkDenseMatrix): FitState
+  def runTest(
+      x: DenseMatrix[Double],
+      y: DenseVector[Double],
+      nullModelFit: Option[FitState]): InternalRow
```

Comment:
I think it's actually not a good idea to capitalize matrix variables since variable capitalization has a semantic in scala: https://www.artima.com/pins1ed/case-classes-and-pattern-matching.html#15.2

```
How does the Scala compiler know that Pi is a constant imported from the java.lang.Math object, and not a variable that stands for the selector value itself? Scala uses a simple lexical rule for disambiguation: a simple name starting with a lowercase letter is taken to be a pattern variable; all other references are taken to be constants.
```

I've seen this cause bugs before.